ons-spark/spark-concepts/partitions.md

```diff
@@ -0,0 +1,941 @@
+```
+
+```{code-tab} r R
+
+repartition_path <- paste0(config$checkpoint_path, "/rescue_by_year.parquet")
+sparklyr::spark_write_parquet(rescue, 
+                              repartition_path,
+                              mode='overwrite',
+                              partition_by='CalYear')
+
+```
+````
+This will create multiple directories in the `rescue_by_year.parquet` directory on the file system, one for each year in the data. 
+
+The easiest way to see this is by navigating to these directories using the file browser in HUE. Alternativly we can use the `subprocess` package to run lines of code through the terminal to return the contents of the `rescue_by_year.parquet` directory.
+````{tabs}
+```{code-tab} py
+import subprocess
+cmd = f"hdfs dfs -ls -C {repartition_path}"
+p = subprocess.run(cmd, shell=True, stdout=subprocess.PIPE, universal_newlines=True)
+print(p.stdout)
+```
+
+```{code-tab} r R
+
+cmd <- paste0("hdfs dfs -rm -r -skipTrash ", repartition_path)
+system(cmd)
+
+```
+````
+
+```plaintext
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2009
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2010
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2011
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2012
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2013
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2014
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2015
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2016
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2017
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2018
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/CalYear=2019
+file:///home/cdsw/ons-spark/checkpoints/rescue_by_year.parquet/_SUCCESS
+```
+On the right of the ouput above you will see there is one directory for each `CalYear`. So to import a subset of the data we can use the specific path for that year or filter the data in Spark and let Spark work out which folders to look for. 
+
+Finally, we will delete these files to clean up the file system.
+````{tabs}
+```{code-tab} py
+cmd = f"hdfs dfs -ls -C {repartition_path}"
+p = subprocess.run(cmd, shell=True, stdout=subprocess.PIPE, universal_newlines=True)
+```
+
+```{code-tab} r R
+
+cmd <- paste0("hdfs dfs -rm -r -skipTrash ", repartition_path)
+system(cmd)
+
+```
+````
+## How should I partition my data?
+
+The short answer is- don't worry about it too much!
+
+Let's flip the question around. What are the consequences of getting the number of partitions "wrong"? Most of the time- nothing. You could spend hours trying to optimise the number of partitions and gain a $x2$ increase in processing speed, but it's likely there are other things you can do that will gain $x10$ or $x100$ increase in processing speed. 
```

Comment:
I'd use English rather than LaTeX here - two times, 10 times, 100 times etc. If using LaTeX I think we want times rather than x?

Reply:
I've changed the wording, worth a quick look to make sure it's ok.